### DISCUSSION TITLE
How to do data binding?

### DISCUSSION THREAD

**Discussion by mwdpb** [canton7/Stylet#215] [Q&A] [answered]
I can't seem to find an example in the manual, but how do we bind data to viewmodel?  I felt this is really a stupid question...

In viewmodel, I have an observable list property ```Tables``` whose item has a property ```DisplayName```.  In CM, you set the control name to your property, does Stylet do the same?  Thanks.

The xaml snippet below doesn't work.  I tried naming datagrid control to "Tables" and it doesn't work either.

![image](https://user-images.githubusercontent.com/55845355/107825002-596c1980-6d50-11eb-9742-eea572862c89.png)

**Comment by mwdpb** [comment, marked as answer]
Actually, I got it to work, same problem as the command guard property: you'll need raise the property change event even it is an ```ObservalbeCollection```.  Do you think it's a bug as ```ObservalbeCollection``` do not need manually raising event in plain MVVM implementation, i.e., without MVVM framework.

**Reply by canton7** [reply]
Stylet doesn't do any name-based bindings: it uses normal WPF bindings.

Without seeing your VMs I can't comment on what your problem might be. ObservableCollection only raises events when items are added or removed, and not when the ObservableCollection property itself is assigned to, or when any of the properties of items in the ObservableCollection are changed: perhaps that's your issue?

I very much doubt your situation works without Stylet, but doesn't work with it. Stylet does not interfere with your bindings.

Please paste a full reproducible sample if you want further help with this.

Thanks!

**Reply by mwdpb** [reply]
Thanks for the detailed explanations.  You are absolutely right!!  I was doing assignment to ObservableCollection variable with Stylet while doing item addition when I say it worked without Stylet.  After seeing your reply, I went changed the code and sure enough it worked!  Lesson learned!

Thank you very much and hats off for your knowledge!
